dbms/src/Storages/DeltaMerge/ColumnFile/ColumnFileTinyVectorIndexReader.cpp

```diff
@@ -0,0 +1,156 @@
+// Copyright 2024 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <IO/Buffer/ReadBufferFromString.h>
+#include <IO/Compression/CompressedReadBuffer.h>
+#include <Storages/DeltaMerge/ColumnFile/ColumnFileDataProvider.h>
+#include <Storages/DeltaMerge/ColumnFile/ColumnFileTiny.h>
+#include <Storages/DeltaMerge/ColumnFile/ColumnFileTinyVectorIndexReader.h>
+#include <Storages/DeltaMerge/Index/VectorIndexCache.h>
+#include <Storages/DeltaMerge/Index/VectorSearchPerf.h>
+
+
+namespace DB::DM
+{
+
+void ColumnFileTinyVectorIndexReader::read(
+    MutableColumnPtr & vec_column,
+    const std::span<const VectorIndexViewer::SearchResult> & selected_rows,
+    size_t start_offset,
+    size_t column_size)
```

Comment:
Optional naming suggestion:

```suggestion
void ColumnFileTinyVectorIndexReader::read(
    MutableColumnPtr & vec_column,
    const std::span<const VectorIndexViewer::SearchResult> & read_rowids,
    size_t rowid_offset,
    size_t read_rows)
```